### ISSUE TITLE
Drupal Apigee API Product RBAC not filtering API Products based on Roles

### ISSUE THREAD

**Issue by urbanenomad** [apigee/apigee-edge-drupal#1098]
## Apigee Info
We have both apigee hybrid and apigee-x instances that we want to support

## Steps to Reproduce
1. Enable apigee API product RBAC
![Screenshot 2024-11-21 at 9 29 54 AM](https://github.com/user-attachments/assets/55665a96-7c76-4c07-be81-82533567c1a2)

2. Configured Custom Attribute on some test apigee products
![Screenshot 2024-11-21 at 9 28 53 AM](https://github.com/user-attachments/assets/2f36654d-5526-4c11-a57d-00bc2f800040)

## Actual Behavior
* Non-admin account that was set to one of the attirbute roles still shows all products even more than if we did just products by visiblity.  It now shows all the products.  
![Screenshot 2024-11-22 at 12 07 00 PM](https://github.com/user-attachments/assets/a31e6fa6-da4a-41c7-b8ea-76188e52a689)

* We also did not enable "Bypass API product access control" nor did we enable "Show API products with missing or empty attribute to everyone"

## Expected Behavior
We would have hoped to see when a developer trys to create an App they only see the API products that have their role in the custom attribute

## Version Info
Drupal 10.3.6

Modules: 
Apigee: 3.0.10
Apigee API Catalog 3.0.8

**Comment by kedarkhaire** [comment]
Hi @urbanenomad 
Thanks for creating a ticket here. We will look into it.
Thanks!

**Comment by urbanenomad** [comment]
Is there any updates on this?

**Comment by urbanenomad** [comment]
Hey so I think I figured out the issue at the following lines.

https://github.com/apigee/apigee-edge-drupal/blob/1e480cfb112892a54f4480565e355c9d53c260ac/modules/apigee_edge_apiproduct_rbac/apigee_edge_apiproduct_rbac.module#L70

the line should read like the following.

$result = AccessResult::forbidden("{$operation} is not allowed on {$entity->label()} API product.");

Same for this line.

https://github.com/apigee/apigee-edge-drupal/blob/1e480cfb112892a54f4480565e355c9d53c260ac/modules/apigee_edge_apiproduct_rbac/apigee_edge_apiproduct_rbac.module#L97

$result = AccessResult::forbidden("{$operation} is not allowed on {$entity->label()} API product.");

This resulted with the correct output on the App Create page.

**Comment by kedarkhaire** [comment]
Hi @urbanenomad 
Thanks for the investigation, Can you please create the PR for the same, we will run the tests and check.
If all good, then we will push the changes in next release.

**Comment by kedarkhaire** [comment]
Hi @urbanenomad 
The PR is merged, once we make a new release, the changes will be present on it.

Thanks!

**Comment by mxr576** [comment]
Would not returning explicit forbidden instead of neutral breaks existing customizations, since usually forbidden can be only overridden from an access alter hook, but existing access hook implementation cannot override it?

**Comment by wkimqc** [comment]
yeah that is what I originally thought as well but since I am explicitly leveraging RBAC in my drupal service I would want the RBAC controls be the highest priority of control.  Plus with Neutral which is what it was before didn't seem to actually block based on roles.

**Comment by mxr576** [comment]
Could be true haven't checked, but core and other contrib modules also use Neutral instead of Forbidden for the above mentioned reason, what only varies is the meaning of neutral is given context (allowed or denied at the end).

**Comment by wkimqc** [comment]
I didn't do a deep dive check on this but did a quick look at product access module and it does use forbidden.

https://github.com/apigee/apigee-edge-drupal/blob/99e56337c8c6d95132fce9d88decba39bddd8d90/apigee_edge.module#L536

And if I am using the RBAC module I really want that to be my final control of access to product assign operation.
